File: Python/Array/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexMap = {}
        for i in range(len(nums)):
            remainder = target - nums[i]
            j = indexMap.get(remainder)
            if (j != None):
                return [i,j]
            indexMap[nums[i]] = i
        return []

File: Java/Array/TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int curr = nums[i];
            if (map.get(target - curr) != null) {
                return new int[]{i, map.get(target - curr)};
            }

            map.put(curr, i);
        }

        return new int[2];
    }
}